// @lds/eden-style-constants/lib/es/index.js
/* -- OTHER COMPANY BRANDS -- */
export var colorFacebook = "#3b5998";
export var colorInstagram = "#3c5a96";
export var colorTwitter = "#00b6f1";
export var colorPinterest = "#cb2027";
export var colorYoutube = "#cd201f";
export var colorOdnok = "#ed812b";
export var colorVk = "#45668e";
/* -- Greys  -- */

export var grey10 = "#d0d3d3"; //quartz

export var grey20 = "#a9adad"; // granite

export var grey30 = "#878a8c"; // slate

export var grey40 = "#53575b"; // onyx

export var grey = grey30;
/* Yellows */

export var yellow10 = "#ffb81c"; // citrine

export var yellow20 = "#f68d2e"; // amber

export var yellow30 = "#d45311"; // topaz

export var yellow40 = "#674730"; // bronzite

export var yellow = yellow30;
/* Reds */

export var red10 = "#fc7473"; // coral

export var red20 = "#bd0057"; // ruby

export var red30 = "#8f124a"; // jasper

export var red40 = "#6e0d33"; // garnet

export var red = red30;
/* Blues */

export var blue10 = "#7de3f4"; // larimar

export var blue20 = "#01b6d1"; // turquiose

export var blue30 = "#006184"; // sapphire

export var blue40 = "#003057"; // azurite

export var blue = blue30;
/* Greens */

export var green10 = "#bed21e"; // olivine

export var green20 = "#6db344"; // beryl

export var green30 = "#318d43"; // emerald

export var green40 = "#235c35"; // jade

export var green = green30;
/* UI SPECIFIC COLOR */

/* -- Util Greys -- */

export var grey2 = "#f7f8f8";
export var grey3 = "#eff0f0"; // onyx10

export var grey5 = "#e0e2e2"; // onyx20

/* grey10 is a thing, but defined above */

export var grey15 = "#bdc0c0"; // onyx40

/* grey20 is a thing, but defined above */

/* grey30 is a thing, but defined above */

/* grey40 is a thing, but defined above */

export var grey60 = "#3a3d40"; // onyx110

export var text120 = "#212225"; // onyxtext

export var black125 = "#0d0f10";
export var black = "#0d0f10";
export var white = "#fff";
/* Interactive Colors */

export var info10 = "#20abd7";
export var info40 = "#177c9c"; // Link / colorLink

export var info60 = "#105970";
export var confirmation10 = "#a3d287"; // colorSuccess

export var confirmation40 = "#74bb49";
export var confirmation60 = "#375a22";
export var warning10 = "#ffd072"; // colorWarn

export var warning40 = "#ffae0c";
export var warning60 = "#8f4200";
export var danger10 = "#fc7473"; // colorError

export var danger40 = "#e63939";
export var danger60 = "#b00504";
/* Util Colors */

export var colorError = red10;
export var colorSuccess = "#a3d287";
export var colorWarn = "#ffd072";
export var colorLink30 = "#177c9c"; // blue 600 (same as info40)

export var colorLink = colorLink30;
export var colorLinkHover = blue20;
/* Highlight Palette */

export var highlightBlue = "#e9f7fc";
export var highlightCyan = "#ddfafb";
export var highlightGreen = "#eff7ea";
export var highlightYellow = "#fff8e9";
export var highlightAmber = "#ffefde";
export var highlightRed = "#fee8e7";
export var highlightMagenta = "#fee8fc";
export var highlightViolet = "#f3e9ff";
/* https = `/style_preview.lds.org/type */

export var sans = "\"Open Sans\", \"Zoram\", \"noto sans\", Helvetica, Arial, sansSerif";
export var serif = "\"McKay\", \"Baskerville\", \"Palatino\", \"Palatino Linotype\", \"Georgia\", \"Times New Roman\", serif";
export var monospace = "\"Courier New\", Courier, monospace";
/* effectively 16px (16px * 100% = 16px) but respects users' special needs */

export var fontSizeRoot = "100%";
/* desired px / 16px (root size) = rem */

export var fontSize11 = "0.6875rem";
export var fontSize13 = "0.8125rem";
export var fontSize14 = "0.875rem";
export var fontSize16 = "1rem";
export var fontSize18 = "1.125rem";
export var fontSize20 = "1.25rem";
export var fontSize28 = "1.75rem";
export var fontSize32 = "2rem";
export var fontSize42 = "2.625rem";
export var lineHeightTight = 1.2;
export var lineHeight = 1.4;
export var lineHeightLoose = 1.6;
/* text_transform variables (may not internationalize so variablelize) */

export var capitalize = "capitalize";
export var uppercase = "uppercase";
/* Spacing Variables  */

export var spacing4 = "0.25rem";
/* 4px */

export var spacing8 = "0.5rem";
/* 8px */

export var spacing16 = "1rem";
/* 16px */

export var spacing32 = "2rem";
/* 32px */

export var spacing64 = "4rem";
/* 64px */

export var spacing128 = "8rem";
/* 128px */

/*
The current LDS header has z-index higher than a thousand. New parts
of the VIO require menus/modals that should cover header/footer so
must add this gross, temporary thing until new pf header/footer are released
(expected 2017)
*/

export var zIndexHigherThanLegacyPlatform = 2000;
/* -- MATERIAL BREAKPOINTS -- */

export var breakWidth480 = "480px";
export var breakWidth600 = "600px";
export var breakWidth840 = "840px";
export var breakWidth960 = "960px";
export var breakWidth1280 = "1280px";
export var breakWidth1440 = "1440px";
export var breakWidth1600 = "1600px";
export var breakWidthMax = breakWidth1600;
/*
According to material design, the gutter width should be:
	16px (1em) when the smallest width of the devices is < 600
	24px (1.5em) otherwise
*/

export var breakWidthGutterShift = breakWidth600; // Approved aspect ratios

export var aspectRatio1By1 = {
  width: 1,
  height: 1,
  key: "1:1"
};
export var aspectRatio5By6 = {
  width: 5,
  height: 6,
  key: "5:6"
};
export var aspectRatio5By8 = {
  width: 5,
  height: 8,
  key: "5:8"
};
export var aspectRatio6By5 = {
  width: 6,
  height: 5,
  key: "6:5"
};
export var aspectRatio8By3 = {
  width: 8,
  height: 3,
  key: "8:3"
};
export var aspectRatio8By5 = {
  width: 8,
  height: 5,
  key: "8:5"
};
export var aspectRatio16By9 = {
  width: 16,
  height: 9,
  key: "16:9"
};
export var aspectRatios = [aspectRatio1By1, aspectRatio5By6, aspectRatio5By8, aspectRatio6By5, aspectRatio8By3, aspectRatio8By5, aspectRatio16By9];
