// @lds/eden-style-constants/src/Details/index.js
import React from "react";
import styled from "styled-components";
import Grid from "@lds/eden-grid";
import * as styleConstants from "@lds/eden-style-constants";

const FormWrapper = styled.div`
  margin-bottom: 15px;
  width: 500px;
`;

const Input = styled.input`
  padding: 5px;
`;

const StyleConstantWrapper = styled.div`
  padding: 15px;
  border: 1px solid #ccc;
  border-radius: 5px;
  display: flex;
  justify-content: space-between;
`;

const StyleConstantNameWrapper = styled.div`
  font-weight: bold;
`;

const StyleConstantValueWrapper = styled.div`
  font-family: monospace;
  font-size: 12px;
`;

const StyleConstantExampleWrapper = styled.div`
  margin-left: 15px;
`;

const ColorTile = styled.div`
  background: ${p => p.color};
  width: 50px;
  height: 50px;
  border-radius: 5px;
  border: 1px solid #eee;
`;

const allStyleConstantKeys = Object.keys(styleConstants);

const colorRegex = /(#(?:[0-9a-f]{2}){2,4}|#[0-9a-f]{3}|(?:rgba?|hsla?)\((?:\d+%?(?:deg|rad|grad|turn)?(?:,|\s)+){2,3}[\s/]*[\d.]+%?\))/i;

class Details extends React.Component {
  state = {
    search: "",
    visibleStyleConstantKeys: allStyleConstantKeys
  };

  handleChange = event => {
    this.setState({
      search: event.target.value,
      visibleStyleConstantKeys: allStyleConstantKeys.filter(styleConstantKey =>
        styleConstantKey
          .toLowerCase()
          .includes(event.target.value.toLowerCase())
      )
    });
  };

  handleSubmit = event => {
    event.preventDefault();
  };

  render() {
    const { visibleStyleConstantKeys } = this.state;

    return (
      <section>
        <FormWrapper>
          <form onSubmit={this.handleSubmit}>
            <label htmlFor="styleConstantSearch">
              <Input
                id="styleConstantSearch"
                type="text"
                placeholder="Search style constants"
                search={this.state.search}
                onChange={this.handleChange}
              />
            </label>
          </form>
        </FormWrapper>

        <Grid gap="16" min="300px">
          {visibleStyleConstantKeys.length > 0 ? (
            visibleStyleConstantKeys.map(styleConstant => {
              const name = `${styleConstant}`;
              const value = styleConstants[styleConstant];

              const isColor = colorRegex.test(value);
              const isObject = typeof value === "object";

              const displayValue = isObject ? (
                <pre>
                  <code>{JSON.stringify(value, null, 2)}</code>
                </pre>
              ) : (
                `${value}`
              );

              return (
                <StyleConstantWrapper key={styleConstant}>
                  <div>
                    <StyleConstantNameWrapper>{name}</StyleConstantNameWrapper>
                    <StyleConstantValueWrapper>
                      {displayValue}
                    </StyleConstantValueWrapper>
                  </div>

                  <StyleConstantExampleWrapper>
                    {isColor ? <ColorTile color={value} /> : null}
                  </StyleConstantExampleWrapper>
                </StyleConstantWrapper>
              );
            })
          ) : (
            <div>No style constants found that match your search</div>
          )}
        </Grid>
      </section>
    );
  }
}

export default Details;
