// @lds/eden-style-constants/lib/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.breakWidthMax = exports.breakWidth1600 = exports.breakWidth1440 = exports.breakWidth1280 = exports.breakWidth960 = exports.breakWidth840 = exports.breakWidth600 = exports.breakWidth480 = exports.zIndexHigherThanLegacyPlatform = exports.spacing128 = exports.spacing64 = exports.spacing32 = exports.spacing16 = exports.spacing8 = exports.spacing4 = exports.uppercase = exports.capitalize = exports.lineHeightLoose = exports.lineHeight = exports.lineHeightTight = exports.fontSize42 = exports.fontSize32 = exports.fontSize28 = exports.fontSize20 = exports.fontSize18 = exports.fontSize16 = exports.fontSize14 = exports.fontSize13 = exports.fontSize11 = exports.fontSizeRoot = exports.monospace = exports.serif = exports.sans = exports.highlightViolet = exports.highlightMagenta = exports.highlightRed = exports.highlightAmber = exports.highlightYellow = exports.highlightGreen = exports.highlightCyan = exports.highlightBlue = exports.colorLinkHover = exports.colorLink = exports.colorLink30 = exports.colorWarn = exports.colorSuccess = exports.colorError = exports.danger60 = exports.danger40 = exports.danger10 = exports.warning60 = exports.warning40 = exports.warning10 = exports.confirmation60 = exports.confirmation40 = exports.confirmation10 = exports.info60 = exports.info40 = exports.info10 = exports.white = exports.black = exports.black125 = exports.text120 = exports.grey60 = exports.grey15 = exports.grey5 = exports.grey3 = exports.grey2 = exports.green = exports.green40 = exports.green30 = exports.green20 = exports.green10 = exports.blue = exports.blue40 = exports.blue30 = exports.blue20 = exports.blue10 = exports.red = exports.red40 = exports.red30 = exports.red20 = exports.red10 = exports.yellow = exports.yellow40 = exports.yellow30 = exports.yellow20 = exports.yellow10 = exports.grey = exports.grey40 = exports.grey30 = exports.grey20 = exports.grey10 = exports.colorVk = exports.colorOdnok = exports.colorYoutube = exports.colorPinterest = exports.colorTwitter = exports.colorInstagram = exports.colorFacebook = void 0;
exports.aspectRatios = exports.aspectRatio16By9 = exports.aspectRatio8By5 = exports.aspectRatio8By3 = exports.aspectRatio6By5 = exports.aspectRatio5By8 = exports.aspectRatio5By6 = exports.aspectRatio1By1 = exports.breakWidthGutterShift = void 0;

/* -- OTHER COMPANY BRANDS -- */
var colorFacebook = "#3b5998";
exports.colorFacebook = colorFacebook;
var colorInstagram = "#3c5a96";
exports.colorInstagram = colorInstagram;
var colorTwitter = "#00b6f1";
exports.colorTwitter = colorTwitter;
var colorPinterest = "#cb2027";
exports.colorPinterest = colorPinterest;
var colorYoutube = "#cd201f";
exports.colorYoutube = colorYoutube;
var colorOdnok = "#ed812b";
exports.colorOdnok = colorOdnok;
var colorVk = "#45668e";
/* -- Greys  -- */

exports.colorVk = colorVk;
var grey10 = "#d0d3d3"; //quartz

exports.grey10 = grey10;
var grey20 = "#a9adad"; // granite

exports.grey20 = grey20;
var grey30 = "#878a8c"; // slate

exports.grey30 = grey30;
var grey40 = "#53575b"; // onyx

exports.grey40 = grey40;
var grey = grey30;
/* Yellows */

exports.grey = grey;
var yellow10 = "#ffb81c"; // citrine

exports.yellow10 = yellow10;
var yellow20 = "#f68d2e"; // amber

exports.yellow20 = yellow20;
var yellow30 = "#d45311"; // topaz

exports.yellow30 = yellow30;
var yellow40 = "#674730"; // bronzite

exports.yellow40 = yellow40;
var yellow = yellow30;
/* Reds */

exports.yellow = yellow;
var red10 = "#fc7473"; // coral

exports.red10 = red10;
var red20 = "#bd0057"; // ruby

exports.red20 = red20;
var red30 = "#8f124a"; // jasper

exports.red30 = red30;
var red40 = "#6e0d33"; // garnet

exports.red40 = red40;
var red = red30;
/* Blues */

exports.red = red;
var blue10 = "#7de3f4"; // larimar

exports.blue10 = blue10;
var blue20 = "#01b6d1"; // turquiose

exports.blue20 = blue20;
var blue30 = "#006184"; // sapphire

exports.blue30 = blue30;
var blue40 = "#003057"; // azurite

exports.blue40 = blue40;
var blue = blue30;
/* Greens */

exports.blue = blue;
var green10 = "#bed21e"; // olivine

exports.green10 = green10;
var green20 = "#6db344"; // beryl

exports.green20 = green20;
var green30 = "#318d43"; // emerald

exports.green30 = green30;
var green40 = "#235c35"; // jade

exports.green40 = green40;
var green = green30;
/* UI SPECIFIC COLOR */

/* -- Util Greys -- */

exports.green = green;
var grey2 = "#f7f8f8";
exports.grey2 = grey2;
var grey3 = "#eff0f0"; // onyx10

exports.grey3 = grey3;
var grey5 = "#e0e2e2"; // onyx20

/* grey10 is a thing, but defined above */

exports.grey5 = grey5;
var grey15 = "#bdc0c0"; // onyx40

/* grey20 is a thing, but defined above */

/* grey30 is a thing, but defined above */

/* grey40 is a thing, but defined above */

exports.grey15 = grey15;
var grey60 = "#3a3d40"; // onyx110

exports.grey60 = grey60;
var text120 = "#212225"; // onyxtext

exports.text120 = text120;
var black125 = "#0d0f10";
exports.black125 = black125;
var black = "#0d0f10";
exports.black = black;
var white = "#fff";
/* Interactive Colors */

exports.white = white;
var info10 = "#20abd7";
exports.info10 = info10;
var info40 = "#177c9c"; // Link / colorLink

exports.info40 = info40;
var info60 = "#105970";
exports.info60 = info60;
var confirmation10 = "#a3d287"; // colorSuccess

exports.confirmation10 = confirmation10;
var confirmation40 = "#74bb49";
exports.confirmation40 = confirmation40;
var confirmation60 = "#375a22";
exports.confirmation60 = confirmation60;
var warning10 = "#ffd072"; // colorWarn

exports.warning10 = warning10;
var warning40 = "#ffae0c";
exports.warning40 = warning40;
var warning60 = "#8f4200";
exports.warning60 = warning60;
var danger10 = "#fc7473"; // colorError

exports.danger10 = danger10;
var danger40 = "#e63939";
exports.danger40 = danger40;
var danger60 = "#b00504";
/* Util Colors */

exports.danger60 = danger60;
var colorError = red10;
exports.colorError = colorError;
var colorSuccess = "#a3d287";
exports.colorSuccess = colorSuccess;
var colorWarn = "#ffd072";
exports.colorWarn = colorWarn;
var colorLink30 = "#177c9c"; // blue 600 (same as info40)

exports.colorLink30 = colorLink30;
var colorLink = colorLink30;
exports.colorLink = colorLink;
var colorLinkHover = blue20;
/* Highlight Palette */

exports.colorLinkHover = colorLinkHover;
var highlightBlue = "#e9f7fc";
exports.highlightBlue = highlightBlue;
var highlightCyan = "#ddfafb";
exports.highlightCyan = highlightCyan;
var highlightGreen = "#eff7ea";
exports.highlightGreen = highlightGreen;
var highlightYellow = "#fff8e9";
exports.highlightYellow = highlightYellow;
var highlightAmber = "#ffefde";
exports.highlightAmber = highlightAmber;
var highlightRed = "#fee8e7";
exports.highlightRed = highlightRed;
var highlightMagenta = "#fee8fc";
exports.highlightMagenta = highlightMagenta;
var highlightViolet = "#f3e9ff";
/* https = `/style_preview.lds.org/type */

exports.highlightViolet = highlightViolet;
var sans = "\"Open Sans\", \"Zoram\", \"noto sans\", Helvetica, Arial, sansSerif";
exports.sans = sans;
var serif = "\"McKay\", \"Baskerville\", \"Palatino\", \"Palatino Linotype\", \"Georgia\", \"Times New Roman\", serif";
exports.serif = serif;
var monospace = "\"Courier New\", Courier, monospace";
/* effectively 16px (16px * 100% = 16px) but respects users' special needs */

exports.monospace = monospace;
var fontSizeRoot = "100%";
/* desired px / 16px (root size) = rem */

exports.fontSizeRoot = fontSizeRoot;
var fontSize11 = "0.6875rem";
exports.fontSize11 = fontSize11;
var fontSize13 = "0.8125rem";
exports.fontSize13 = fontSize13;
var fontSize14 = "0.875rem";
exports.fontSize14 = fontSize14;
var fontSize16 = "1rem";
exports.fontSize16 = fontSize16;
var fontSize18 = "1.125rem";
exports.fontSize18 = fontSize18;
var fontSize20 = "1.25rem";
exports.fontSize20 = fontSize20;
var fontSize28 = "1.75rem";
exports.fontSize28 = fontSize28;
var fontSize32 = "2rem";
exports.fontSize32 = fontSize32;
var fontSize42 = "2.625rem";
exports.fontSize42 = fontSize42;
var lineHeightTight = 1.2;
exports.lineHeightTight = lineHeightTight;
var lineHeight = 1.4;
exports.lineHeight = lineHeight;
var lineHeightLoose = 1.6;
/* text_transform variables (may not internationalize so variablelize) */

exports.lineHeightLoose = lineHeightLoose;
var capitalize = "capitalize";
exports.capitalize = capitalize;
var uppercase = "uppercase";
/* Spacing Variables  */

exports.uppercase = uppercase;
var spacing4 = "0.25rem";
/* 4px */

exports.spacing4 = spacing4;
var spacing8 = "0.5rem";
/* 8px */

exports.spacing8 = spacing8;
var spacing16 = "1rem";
/* 16px */

exports.spacing16 = spacing16;
var spacing32 = "2rem";
/* 32px */

exports.spacing32 = spacing32;
var spacing64 = "4rem";
/* 64px */

exports.spacing64 = spacing64;
var spacing128 = "8rem";
/* 128px */

/*
The current LDS header has z-index higher than a thousand. New parts
of the VIO require menus/modals that should cover header/footer so
must add this gross, temporary thing until new pf header/footer are released
(expected 2017)
*/

exports.spacing128 = spacing128;
var zIndexHigherThanLegacyPlatform = 2000;
/* -- MATERIAL BREAKPOINTS -- */

exports.zIndexHigherThanLegacyPlatform = zIndexHigherThanLegacyPlatform;
var breakWidth480 = "480px";
exports.breakWidth480 = breakWidth480;
var breakWidth600 = "600px";
exports.breakWidth600 = breakWidth600;
var breakWidth840 = "840px";
exports.breakWidth840 = breakWidth840;
var breakWidth960 = "960px";
exports.breakWidth960 = breakWidth960;
var breakWidth1280 = "1280px";
exports.breakWidth1280 = breakWidth1280;
var breakWidth1440 = "1440px";
exports.breakWidth1440 = breakWidth1440;
var breakWidth1600 = "1600px";
exports.breakWidth1600 = breakWidth1600;
var breakWidthMax = breakWidth1600;
/*
According to material design, the gutter width should be:
	16px (1em) when the smallest width of the devices is < 600
	24px (1.5em) otherwise
*/

exports.breakWidthMax = breakWidthMax;
var breakWidthGutterShift = breakWidth600; // Approved aspect ratios

exports.breakWidthGutterShift = breakWidthGutterShift;
var aspectRatio1By1 = {
  width: 1,
  height: 1,
  key: "1:1"
};
exports.aspectRatio1By1 = aspectRatio1By1;
var aspectRatio5By6 = {
  width: 5,
  height: 6,
  key: "5:6"
};
exports.aspectRatio5By6 = aspectRatio5By6;
var aspectRatio5By8 = {
  width: 5,
  height: 8,
  key: "5:8"
};
exports.aspectRatio5By8 = aspectRatio5By8;
var aspectRatio6By5 = {
  width: 6,
  height: 5,
  key: "6:5"
};
exports.aspectRatio6By5 = aspectRatio6By5;
var aspectRatio8By3 = {
  width: 8,
  height: 3,
  key: "8:3"
};
exports.aspectRatio8By3 = aspectRatio8By3;
var aspectRatio8By5 = {
  width: 8,
  height: 5,
  key: "8:5"
};
exports.aspectRatio8By5 = aspectRatio8By5;
var aspectRatio16By9 = {
  width: 16,
  height: 9,
  key: "16:9"
};
exports.aspectRatio16By9 = aspectRatio16By9;
var aspectRatios = [aspectRatio1By1, aspectRatio5By6, aspectRatio5By8, aspectRatio6By5, aspectRatio8By3, aspectRatio8By5, aspectRatio16By9];
exports.aspectRatios = aspectRatios;
