// @lds/eden-style-constants/src/index.js
/* -- OTHER COMPANY BRANDS -- */
export const colorFacebook = "#3b5998";
export const colorInstagram = "#3c5a96";
export const colorTwitter = "#00b6f1";
export const colorPinterest = "#cb2027";
export const colorYoutube = "#cd201f";
export const colorOdnok = "#ed812b";
export const colorVk = "#45668e";

/* -- Greys  -- */
export const grey10 = "#d0d3d3"; //quartz
export const grey20 = "#a9adad"; // granite
export const grey30 = "#878a8c"; // slate
export const grey40 = "#53575b"; // onyx

export const grey = grey30;

/* Yellows */
export const yellow10 = "#ffb81c"; // citrine
export const yellow20 = "#f68d2e"; // amber
export const yellow30 = "#d45311"; // topaz
export const yellow40 = "#674730"; // bronzite

export const yellow = yellow30;

/* Reds */
export const red10 = "#fc7473"; // coral
export const red20 = "#bd0057"; // ruby
export const red30 = "#8f124a"; // jasper
export const red40 = "#6e0d33"; // garnet

export const red = red30;

/* Blues */
export const blue10 = "#7de3f4"; // larimar
export const blue20 = "#01b6d1"; // turquiose
export const blue30 = "#006184"; // sapphire
export const blue40 = "#003057"; // azurite

export const blue = blue30;

/* Greens */
export const green10 = "#bed21e"; // olivine
export const green20 = "#6db344"; // beryl
export const green30 = "#318d43"; // emerald
export const green40 = "#235c35"; // jade

export const green = green30;

/* UI SPECIFIC COLOR */

/* -- Util Greys -- */
export const grey2 = "#f7f8f8";
export const grey3 = "#eff0f0"; // onyx10
export const grey5 = "#e0e2e2"; // onyx20
/* grey10 is a thing, but defined above */
export const grey15 = "#bdc0c0"; // onyx40
/* grey20 is a thing, but defined above */
/* grey30 is a thing, but defined above */
/* grey40 is a thing, but defined above */
export const grey60 = "#3a3d40"; // onyx110
export const text120 = "#212225"; // onyxtext
export const black125 = "#0d0f10";
export const black = "#0d0f10";
export const white = "#fff";

/* Interactive Colors */
export const info10 = "#20abd7";
export const info40 = "#177c9c"; // Link / colorLink
export const info60 = "#105970";
export const confirmation10 = "#a3d287"; // colorSuccess
export const confirmation40 = "#74bb49";
export const confirmation60 = "#375a22";
export const warning10 = "#ffd072"; // colorWarn
export const warning40 = "#ffae0c";
export const warning60 = "#8f4200";
export const danger10 = "#fc7473"; // colorError
export const danger40 = "#e63939";
export const danger60 = "#b00504";

/* Util Colors */
export const colorError = red10;
export const colorSuccess = "#a3d287";
export const colorWarn = "#ffd072";

export const colorLink30 = "#177c9c"; // blue 600 (same as info40)
export const colorLink = colorLink30;
export const colorLinkHover = blue20;

/* Highlight Palette */
export const highlightBlue = "#e9f7fc";
export const highlightCyan = "#ddfafb";
export const highlightGreen = "#eff7ea";
export const highlightYellow = "#fff8e9";
export const highlightAmber = "#ffefde";
export const highlightRed = "#fee8e7";
export const highlightMagenta = "#fee8fc";
export const highlightViolet = "#f3e9ff";

/* https = `/style_preview.lds.org/type */
export const sans = `"Open Sans", "Zoram", "noto sans", Helvetica, Arial, sansSerif`;
export const serif = `"McKay", "Baskerville", "Palatino", "Palatino Linotype", "Georgia", "Times New Roman", serif`;
export const monospace = `"Courier New", Courier, monospace`;

/* effectively 16px (16px * 100% = 16px) but respects users' special needs */
export const fontSizeRoot = `100%`;

/* desired px / 16px (root size) = rem */
export const fontSize11 = "0.6875rem";
export const fontSize13 = "0.8125rem";
export const fontSize14 = "0.875rem";
export const fontSize16 = "1rem";
export const fontSize18 = "1.125rem";
export const fontSize20 = "1.25rem";
export const fontSize28 = "1.75rem";
export const fontSize32 = "2rem";
export const fontSize42 = "2.625rem";

export const lineHeightTight = 1.2;
export const lineHeight = 1.4;
export const lineHeightLoose = 1.6;

/* text_transform variables (may not internationalize so variablelize) */
export const capitalize = `capitalize`;
export const uppercase = `uppercase`;

/* Spacing Variables  */
export const spacing4 = "0.25rem"; /* 4px */
export const spacing8 = "0.5rem"; /* 8px */
export const spacing16 = "1rem"; /* 16px */
export const spacing32 = "2rem"; /* 32px */
export const spacing64 = "4rem"; /* 64px */
export const spacing128 = "8rem"; /* 128px */

/*
The current LDS header has z-index higher than a thousand. New parts
of the VIO require menus/modals that should cover header/footer so
must add this gross, temporary thing until new pf header/footer are released
(expected 2017)
*/
export const zIndexHigherThanLegacyPlatform = 2000;

/* -- MATERIAL BREAKPOINTS -- */
export const breakWidth480 = "480px";
export const breakWidth600 = "600px";
export const breakWidth840 = "840px";
export const breakWidth960 = "960px";
export const breakWidth1280 = "1280px";
export const breakWidth1440 = "1440px";
export const breakWidth1600 = "1600px";

export const breakWidthMax = breakWidth1600;

/*
According to material design, the gutter width should be:
	16px (1em) when the smallest width of the devices is < 600
	24px (1.5em) otherwise
*/
export const breakWidthGutterShift = breakWidth600;

// Approved aspect ratios
export const aspectRatio1By1 = { width: 1, height: 1, key: "1:1" };
export const aspectRatio5By6 = { width: 5, height: 6, key: "5:6" };
export const aspectRatio5By8 = { width: 5, height: 8, key: "5:8" };
export const aspectRatio6By5 = { width: 6, height: 5, key: "6:5" };
export const aspectRatio8By3 = { width: 8, height: 3, key: "8:3" };
export const aspectRatio8By5 = { width: 8, height: 5, key: "8:5" };
export const aspectRatio16By9 = { width: 16, height: 9, key: "16:9" };

export const aspectRatios = [
  aspectRatio1By1,
  aspectRatio5By6,
  aspectRatio5By8,
  aspectRatio6By5,
  aspectRatio8By3,
  aspectRatio8By5,
  aspectRatio16By9
];
